### PR TITLE
 #157 finalize v2 spark infrastructure

### PR DESCRIPTION
**Pull request by cwoods-cpointe**
Default all new applications to use spark-infrastructure to the V2 charts. Mark any V1 users with a depreciation warning. Provide migration instructions for going from V1 -> V2.

Update spark-events to be saved to a more dynamic PVC instead of a S3 bucket.

Signed-off-by: Peter McClonski <mcclonski_peter@bah.com>

 #300 Resolve several issues with the V2 charts

Correct the Spark-event PV host node mount point. Update the spark configmap name to be in line with our documentaion (-conf -> -config).

Corrected a bug in the Configuration-store PVC being writable instead of read only.

### CODE REVIEW

File: docs/modules/ROOT/pages/containers.adoc [hunk 1]
```diff
@@ -189,16 +189,17 @@ TIP: If your project is under version control, we recommend using a diff tool to
...
-|Policy Decision Point                  | `policy-decision-point-deploy-v2`    |https://github.com/boozallen/aissemble/blob/{git-tree}/extensions/extensions-helm/aissemble-policy-decision-point-chart/README.md[Policy Decision Point,role=external,window=_blank]
-|S3 Local                               | `s3local-deploy-v2`                  |https://github.com/boozallen/aissemble/blob/{git-tree}/extensions/extensions-helm/aissemble-localstack-chart/README.md[S3 Local,role=external,window=_blank]
-|Configuration Store                    | `configuration-store-deploy-v2`      |https://github.com/boozallen/aissemble/blob/{git-tree}/extensions/extensions-helm/aissemble-configuration-store-chart/README.md[Configuration Store,role=external,window=_blank]
-|Spark Operator                         | `aissemble-spark-operator-deploy-v2` |https://github.com/boozallen/aissemble/blob/{git-tree}/extensions/extensions-helm/aissemble-spark-operator-chart/README.md[Spark Operator,role=external,window=_blank]
+|Application                            | Fermenter Profile                         | ReadMe
```
Comment (cwoods-cpointe) [inline docs/modules/ROOT/pages/containers.adoc:192]: i: Only code format changes

File: extensions/extensions-helm/aissemble-mlflow-chart/README.md [hunk 2]
```diff
@@ -11,61 +11,61 @@ helm install mlflow oci://ghcr.io/boozallen/aissemble-mlflow-chart --version <AI
 # Properties
 The following properties are inherited from the base [MLflow chart](https://github.com/bitnami/charts/blob/main/bitnami/mlflow/README.md), but with updated default values. 
 
-| Property | Description | Default |
```
Comment (cwoods-cpointe) [inline extensions/extensions-helm/aissemble-mlflow-chart/README.md:14]: i: Only code format changes

File: extensions/extensions-helm/aissemble-spark-application-chart/values.yaml [hunk 3]
```diff
@@ -21,6 +30,9 @@ sparkApp:
       - name: ivy-cache
         persistentVolumeClaim:
           claimName: ivy-cache
+      - name: spark-events
```
Comment (cwoods-cpointe) [inline extensions/extensions-helm/aissemble-spark-application-chart/values.yaml:33]: I: Spark events were previously saved to s3. Changed to save to a PVC

File: extensions/extensions-helm/aissemble-spark-operator-chart/README.md [hunk 4]
```diff
@@ -37,17 +37,17 @@ aissemble-spark-operator-chart:
     webhook.enable: false
 ```
 
-| Property                           | Description                        | Required Override | Default                                                           |
```
Comment (cwoods-cpointe) [inline extensions/extensions-helm/aissemble-spark-operator-chart/README.md:40]: i: Only code format changes

File: foundation/foundation-mda/src/main/resources/templates/deployment/configuration-store/v2/configuration-store.values.yaml.vm [hunk 5]
```diff
@@ -18,10 +18,10 @@ aissemble-configuration-store-chart:
         - name: configurations
           persistentVolumeClaim:
             claimName: configuration-store-claim
-          readOnly: true
       supplementalVolumeMounts:
         - name: configurations
           mountPath: "/configurations"
+          readOnly: true
```
Comment (cwoods-cpointe) [inline foundation/foundation-mda/src/main/resources/templates/deployment/configuration-store/v2/configuration-store.values.yaml.vm:24]: i: resolved issue where the configuration-store PVC was not read only

File: extensions/extensions-helm/aissemble-spark-application-chart/values.yaml (hunk 3 above)
Comment (carter-cundiff) [reply extensions/extensions-helm/aissemble-spark-application-chart/values.yaml:33]: Should we include this new property in the README properties section?

**Review by JeffreyRoss — APPROVED** [review]

File: extensions/extensions-helm/aissemble-spark-application-chart/values.yaml (hunk 3 above)
Comment (cwoods-cpointe) [reply extensions/extensions-helm/aissemble-spark-application-chart/values.yaml:33]: Good question. There is a section on the shared Ivy cache which is similarly a pvc. I can add a section to the readme for Spark-events

File: extensions/extensions-helm/extensions-helm-spark-infrastructure/aissemble-spark-history-chart/README.md [hunk 6]
```diff
@@ -44,36 +51,58 @@ _**NOTE:**_ *the version should match the aiSSEMBLE project version.*
...
+existing customizations from the old chart to the new v2 chart.
+
+## Property Location
+All properties listed below have been moved to the parent chart. If any properties are set to the default value, we 
+recommend removing them from your values file entirely.
+
+**Note**: *all new property locations include the prefix `aissemble-spark-history-chart.`*
+
+## Property Removed
+The following properties no longer exist.
```
Comment (carter-cundiff) [inline extensions/extensions-helm/extensions-helm-spark-infrastructure/aissemble-spark-history-chart/README.md:108]: Should there be a table within these sections with the properties?

File: extensions/extensions-helm/extensions-helm-spark-infrastructure/aissemble-spark-history-chart/README.md [hunk 7]
```diff
@@ -44,36 +51,58 @@ _**NOTE:**_ *the version should match the aiSSEMBLE project version.*
...
+the [Pod Security admission controller](https://kubernetes.io/docs/concepts/security/pod-security-admission/)
+to allow hostPath volumes, either across the board or for Spark History Server pods specifically. Alternatively,
+a [`local`
 volume](https://kubernetes.io/docs/concepts/storage/volumes/#local) can be used which alleviates most of the security
 concerns as well as the single-node limitation. However, redeployments of the Spark Operator will not reuse the same
 volume, meaning the cache will be lost between deployments.
+
+# Migration from aiSSEMBLE v1 Helm Charts
+
+If you are migrating from the v1 version of the hive-metastore-service chart, use the tables below to apply any
```
Comment (carter-cundiff) [inline extensions/extensions-helm/extensions-helm-spark-infrastructure/aissemble-spark-history-chart/README.md:98]: This should refer to the v1 spark-infrastructure chart

Comment (cwoods-cpointe) [reply extensions/extensions-helm/extensions-helm-spark-infrastructure/aissemble-spark-history-chart/README.md:98]: yes

File: extensions/extensions-helm/extensions-helm-spark-infrastructure/aissemble-spark-history-chart/README.md (hunk 6 above)
Comment (cwoods-cpointe) [reply extensions/extensions-helm/extensions-helm-spark-infrastructure/aissemble-spark-history-chart/README.md:108]: There were no removed properties but left the header to follow the convention. Can remove if we think its confusing.

Comment (carter-cundiff) [reply extensions/extensions-helm/extensions-helm-spark-infrastructure/aissemble-spark-history-chart/README.md:108]: I think we should keep the sections to remain the same as the other README's but maybe mention that's there's just no properties in this instance. For example in the section above `All properties listed below have been moved to the parent chart` makes it seem like we just forgot to list the properties.

**Review by carter-cundiff — APPROVED** [review]

File: extensions/extensions-helm/extensions-helm-spark-infrastructure/aissemble-spark-history-chart/README.md (hunk 6 above)
Comment (cwoods-cpointe) [reply extensions/extensions-helm/extensions-helm-spark-infrastructure/aissemble-spark-history-chart/README.md:108]: I like it! Ill update it to include the section but give context as to why its blank

**Pull request merged**